assists/baremetalconfig_xlnx.py

```diff
@@ -0,0 +1,292 @@
+            reg = value[cells * idx + 1]
+
+        size1 = value[cells * idx + na]
+        if size1 != '0x0':
+            val = value[cells * idx + ns + 1]
+            size = size1 + val
+        else:
+            size = value[cells * idx + ns + 1]
+    elif cells == 2:
+        reg = value[idx * cells]
+        size = value[idx * cells + 1]
+    else:
+        reg = value[0]
+    return reg, size
+
+def get_interrupt_prop(fdt, node, value):
+    intr = []
+    inp = Lopper.property_get(fdt, node, 'interrupt-parent')
+    intr_parent = fdt.node_offset_by_phandle(inp)
+    inc = Lopper.property_get(fdt, intr_parent, '#interrupt-cells')
+    """
+    Baremetal Interrupt Property format:
+        bits[11:0]  interrupt-id
+        bits[15:12] trigger type and level flags
+    """
+    # Below logic converts the interrupt propery value to baremetal
+    # interrupt property format.
+    nintr = len(value)/inc
+    tmp = inc % 2
+    for val in range(0, int(nintr)):
+        intr_sensitivity = value[tmp+1] << 12
+        intr_id = value[tmp]
+        intr.append(hex(intr_id + intr_sensitivity))
+
+    return intr
+
+#Return the base address of the interrupt parent.
+def get_intrerrupt_parent(fdt, node, value):
+    intr_parent = fdt.node_offset_by_phandle(value[0])
+    addr = Lopper.property_get(fdt, intr_parent, 'reg', LopperFmt.COMPOUND)
+    reg,size = scan_reg_size(fdt, intr_parent, addr, 0)
+    return reg
+
+class DtbtoCStruct(object):
+    def __init__(self, out_file):
+        self._outfile = open(out_file, 'w')
+        self._lines = []
+
+    def out(self, line):
+        """Output a string to the output file
+
+        Args:
+            line: String to output
+        """
+        self._outfile.write(line)
+
+    def buf(self, line):
+        """Buffer up a string to send later
+
+        Args:
+            line: String to add to our 'buffer' list
+        """
+        self._lines.append(line)
+
+    def get_buf(self):
+        """Get the contents of the output buffer, and clear it
+
+        Returns:
+            The output buffer, which is then cleared for future use
+        """
+        lines = self._lines
+        self._lines = []
+        return lines
+
+def is_compat(node, compat_string_to_test):
+    if re.search( "baremetaldrv,config", compat_string_to_test):
+        return xlnx_generate_bm_config
+    return ""
+
+# tgt_node: is the baremetal config top level domain node number
+# sdt: is the system device-tree
+# options: baremetal driver meta-data file path
+def xlnx_generate_bm_config(tgt_node, sdt, options):
+    root_node = sdt.tree[tgt_node]
+    root_sub_nodes = root_node.subnodes()
+    node_list = []
+    # Traverse the tree and find the nodes having status=ok property
+    for node in root_sub_nodes:
+        try:
+            status = node["status"].value
+            if "okay" in status:
+                node_list.append(node)
+        except:
+           pass
+
+    sdt.FDT = Lopper.dt_to_fdt(sdt.dtb, 'rb')
```

Comment:
You shouldn't need to do this. The sdt that is passed into the callback already has the dtb read, and assigned to sdt.FDT. That is the core system device tree.

Reply:
I am getting sdt.FDT as null when my assist got called that's why i am trying to get the FDT again...

Reply:
Hmmm. That's almost impossible to happen, since things will immediately crash. Can you supply me a test command line ? That way I can debug it here ?

Reply:
Yes this line is not needed , i am not able to reproduce the issue that i have mentioned above, the issue that i observed may be due to a race in my script after cleanup things started working :) sorry for the noise.